cllc-public-app/ClientApp/src/app/services/special-events-data.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from "@angular/core";
+import { HttpClient } from "@angular/common/http";
+import { catchError } from "rxjs/operators";
+import { Observable } from "rxjs";
+import { DataService } from "./data.service";
+import { SepApplication } from "@models/sep-application.model";
+
+@Injectable()
+export class SpecialEventsDataService extends DataService {
+
+  constructor(private http: HttpClient) {
+    super();
+  }
+
+
+  getSpecialEvent(id: string): Observable<SepApplication[]> {
+    const apiPath = `api/special-events/${id}`;
+    return this.http.get<SepApplication[]>(apiPath, { headers: this.headers })
+      .pipe(catchError(this.handleError));
+  }
+
+  /**
+   * Create a new special event application in Dynamics
+   * @param data - special event application data
+   */
+  createSepApplication(data: SepApplication) {
+    return this.http.post<SepApplication>("api/special-events/", data, { headers: this.headers })
+      .pipe(catchError(this.handleError));
+  }
+
+  /**
+   * update a  special event application in Dynamics
+   * @param data - special event application data
+   */
+  updateSepApplication(data: SepApplication, id: string) {
+    return this.http.put<SepApplication>(`api/special-events/${id}`, data, { headers: this.headers })
+      .pipe(catchError(this.handleError));
+  }
+
+  /**
+   * delete a  special event application in Dynamics
+   * @param data - special event application data
+   */
+  deleteSepApplication(id: string) {
+    return this.http.post<SepApplication>(`api/special-events/${id}/delete`, {}, { headers: this.headers })
+      .pipe(catchError(this.handleError));
+  }
```

Comment:
I understand this is partial work and more will come but please note that the **delete** endpoint is not yet implemented in `SpecialEventsController`